test/parallel/test-fs-watch-enoent.js

```diff
@@ -1,21 +1,56 @@
 'use strict';
+
+// This verifies the error thrown by fs.watch.
+
 const common = require('../common');
 const assert = require('assert');
 const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+const path = require('path');
+const nonexistentFile = path.join(tmpdir.path, 'non-existent');
+const uv = process.binding('uv');
+
+tmpdir.refresh();
+
+{
+  const validateError = (err) => {
+    assert.strictEqual(nonexistentFile, err.path);
+    assert.strictEqual(nonexistentFile, err.filename);
+    assert.strictEqual(
+      err.message,
+      `ENOENT: no such file or directory, watch '${nonexistentFile}'`);
+    assert.strictEqual(err.errno, uv.UV_ENOENT);
+    assert.strictEqual(err.code, 'ENOENT');
+    assert.strictEqual(err.syscall, 'watch');
+    return true;
+  };
```

Comment:
As a suggestion: `assert.throws` supports the object notation. So instead of using the function here, it could also be written as:

```js
assert.throws(
  () => assert.fail(),
  {
    path: 'foo',
    filename: 'bar',
    ...
  }
)
```

Reply:
With functions it will be easier to adjust the validations to different platforms, e.g. some throws ENOENT while some throws ENODEV (AIX)

Reply:
I am not sure in what way that is easier to validate but it is only a suggestion, so *shrug*.

Reply:
@BridgeAR  It can be written like this

```
if (err.code === 'ENOENT') {
   // validate ENOENT
} else {
  // validate ENODEV
}
```